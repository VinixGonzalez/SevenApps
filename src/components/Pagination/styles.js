import styled from "styled-components";

export const Content = styled.div`
  .pagination {
    display: flex;
    margin: 20px auto;

    li {
      & + .active {
        background: #15095b;

        a {
          color: #fff;
        }
      }

      &:nth-child(1) {
        background: #0097b020;
      }
      &:nth-child(2) {
        background: #0097b020;
      }
      &:last-child {
        background: #0097b020;
      }
      &:nth-last-child(2) {
        background: #0097b020;
      }

      border: 1px solid #0097b0;
      transition: all 0.5s ease;
      background: #fff;
      border-radius: 50%;
      width: 35px;
      height: 35px;
      margin: 0 10px;

      display: flex;
      align-items: center;
      justify-content: center;

      cursor: pointer;

      &:hover {
        background: #15095b90;
        a {
          color: #fff;
        }
      }

      a {
        font-size: 12px;
        color: #0097b0;
        text-decoration: none;
      }
    }
  }
`;
