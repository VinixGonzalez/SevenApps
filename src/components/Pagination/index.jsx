import React, { useLayoutEffect, useState, memo } from "react";

import Pagination from "react-js-pagination";
import { Content } from "./styles";

const CustomPagination = ({
  usersPerPage,
  totalUsers,
  currentPage,
  handlePageChange,
}) => {
  const useWindowSize = () => {
    const [size, setSize] = useState([0, 0]);
    useLayoutEffect(() => {
      function updateSize() {
        setSize([window.innerWidth, window.innerHeight]);
      }
      window.addEventListener("resize", updateSize);
      updateSize();
      return () => window.removeEventListener("resize", updateSize);
    }, []);
    return size;
  };

  const pageNumbers = [];

  const [width] = useWindowSize();

  for (let i = 1; i <= Math.ceil(totalUsers / usersPerPage); i++) {
    pageNumbers.push(i);
  }

  return (
    <Content currentPage={currentPage} lastPage={pageNumbers.length}>
      <Pagination
        activePage={currentPage || 1}
        itemsCountPerPage={usersPerPage}
        totalItemsCount={totalUsers}
        pageRangeDisplayed={width < 1100 ? 1 : 11}
        onChange={handlePageChange}
      />
    </Content>
  );
};

export default memo(CustomPagination);
