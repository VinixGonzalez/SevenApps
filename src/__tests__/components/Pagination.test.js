import * as React from "react";
import { shallow } from "enzyme";
import Pagination from "../../components/Footer";

const props = {
  onChange: jest.fn(),
  activePage: 1,
  itemsCountPerPage: 20,
  totalItemsCount: 500,
  pageRangeDisplayed: 11,
  title: "pagination",
};

test("should render the pagination given the props", () => {
  const component = shallow(<Pagination {...props} />);

  expect(component).toMatchSnapshot();
});
