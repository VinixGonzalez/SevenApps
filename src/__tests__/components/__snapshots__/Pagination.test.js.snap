// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the pagination given the props 1`] = `
<styled.footer>
  <span
    aria-label="smile"
    role="img"
  >
    made by VinÃ­cius Gonzalez ðŸ¤“ - 2020
  </span>
</styled.footer>
`;
